hedera-node/src/main/java/com/hedera/services/ledger/interceptors/AccountsCommitInterceptor.java

```diff
@@ -41,9 +49,26 @@
  */
 public class AccountsCommitInterceptor implements CommitInterceptor<AccountID, MerkleAccount, AccountProperty> {
 	private final SideEffectsTracker sideEffectsTracker;
+	private final Supplier<MerkleNetworkContext> networkCtx;
+	private final Supplier<MerkleMap<EntityNum, MerkleStakingInfo>> stakingInfo;
+	private final GlobalDynamicProperties dynamicProperties;
+	private boolean rewardsActivated;
+	private boolean rewardBalanceChanged;
+	private long newRewardBalance;
```

Comment:
We could just reset this field to `-1` at the beginning of each call to `preview()`; and then simplify `shouldActivateStakingRewards()` to,
```
	public boolean shouldActivateStakingRewards() {
		return !rewardsActivated && newRewardBalance >= dynamicProperties.getStakingStartThreshold();
	}
```

Reply:
Fixed !